<launch>

<arg name="robot" default="default_robot" />
<arg name="robot_id" default="1" />
<arg name="sensor_update_rate" default="10.0" />

<arg name="models_dir" default="$(find multibot_config)/robots" /> 
<arg name="config_dir" default="$(find robot_driver)/share" />

<arg name="wheel_dist" default="0.48" />
<arg name="wheel_radius" default="0.11" />

<!-- Load URDF robot model -->
<param name="$(arg robot)/robot_description"
	command="$(find xacro)/xacro
	--inorder '$(find multibot_config)/robots/yarp4.urdf.xacro'" />

<node name="joint_state_publisher" pkg="joint_state_publisher"
	type="joint_state_publisher"
	ns="$(arg robot)" />
<node name="robot_state_publisher" pkg="robot_state_publisher"
	type="robot_state_publisher"
	ns="$(arg robot)" />

<!-- Debug: place robot in the world by attaching to the map frame -->
<!--node pkg="tf" type="static_transform_publisher"
	 name="static_ded_broadcaster"
	 args="0 0 0 0 0 0 map base_link 100" /-->

<!-- Odom coordinates frame -->
<node pkg="tf" type="static_transform_publisher"
	name="static_broadcaster" args="0 0 0 0 0 0 map odom 100"
	ns="$(arg robot)" />

<node name="robot_driver" pkg="robot_driver" type="robot_driver"
	output="screen" ns="$(arg robot)" clear_params="true"
	respawn="false" required="false"
	launch-prefix="" >
	<param name="sensor_update_rate" type="double" 
		value="$(arg sensor_update_rate)" />
	<param name="robot_id" type="int" value="$(arg robot_id)" />
	<param name="wheel_dist" type="double" value="$(arg wheel_dist)" />
	<param name="wheel_radius" type="double" 
		value="$(arg wheel_radius)" />
	<remap from="ans_topic" to="/ardans_topic" />
	<remap from="action_topic" to="/actions_topic" />
</node>

<node name="obstacle_detector" pkg="obstacle_detector"
	type="obstacle_detector"
	output="log" clear_params="true" ns="$(arg robot)"
	respawn="false" required="false"
	launch-prefix="" >
</node>

<!-- YARP-2 -->
<node name="odometry_encoders" pkg="odometry_encoders"
	type="encoders_odom.py"
	output="log" ns="$(arg robot)" clear_params="true"
	respawn="false" required="false"
	launch-prefix="" >
	<param name="base_link_frame" type="str" value="base_link" />
	<param name="odom_frame" type="str" value="odom" />
	<param name="wheel_dist" type="double" value="$(arg wheel_dist)" />
	<param name="wheel_radius" type="double" 
		value="$(arg wheel_radius)" />
	<param name="ticks_per_revolution" type="int" value="55" />
	<param name="enc_left_min" type="int" value="0" />
	<param name="enc_left_max" type="int" value="65535" />
	<param name="enc_right_min" type="int" value="0" />
	<param name="enc_right_max" type="int" value="65535" />
</node>

</launch>
